### PR TITLE
feat: pending tx ws

### PR DESCRIPTION
**Pull request by lljxx1**
# Pull Request Template

## Database Migration Strategy

For tables such as `orders`, `tokens`, `collections`, etc., we split the deployment into two distinct phases:

1. **PR for Applying Migration Changes:**

   - This PR should solely focus on the database schema changes.
   - Ensure that the migration script is thoroughly tested and is backward compatible.

2. **PR for Applying Code Changes:**
   - This PR should include the code changes that utilize the new database schema.
   - Make sure to handle any new database schema elements within the code.

## Handling New Columns with Default Values

When adding new columns with a default value, follow this approach:

- **Initial Migration Without a Default Value:**

  - Perform the initial migration without setting a default value for the new column.
  - Ensure that the application code is equipped to handle the absence of a default value.

- **Code Modifications:**
  - Adapt the application code to manage the logic around the new column.
  - Once the code is deployed and stable, a subsequent migration can introduce the default value if necessary.

## Additional Notes

Include any additional information that you believe is important for the reviewers to know.

### CODE REVIEW

File: packages/indexer/src/jobs/websocket-events/utils.ts [hunk 1]
```diff
@@ -1,5 +1,29 @@
+import { idb } from "@/common/db";
+import { toBuffer } from "@/common/utils";
+
 // Utility functions for parsing cdc event data
 
+export async function getTokenMetadata(tokenId: string, contract: string) {
```
Comment (nofir) [inline packages/indexer/src/jobs/websocket-events/utils.ts:6]: Is there a reason to create a new method and not use [this](https://github.com/reservoirprotocol/indexer/blob/9965b8b1b7f0f6f7ffc242f5fd7297afadb8dd0e/packages/indexer/src/models/tokens/index.ts#L27)?

File: packages/indexer/src/tests/pending-state/pending.test.ts [hunk 2]
```diff
@@ -22,15 +21,16 @@ describe("PendingState", () => {
       // console.log("pendingTokens", pendingTokens);
       if (!pendingTokens?.length) continue;
       const contract = pendingTokens[0].contract;
-      const pendingTokenIdsBefore = await getContractPendingTokenIds(contract);
-      const recent = await getRecentPendingTokens();
+      const pendingTokenIdsBefore = await getPendingItems(contract);
+      const recent = await getPendingItems();
+      // console.log("recent", recent);
```
Comment (nofir) [inline packages/indexer/src/tests/pending-state/pending.test.ts:26]: please remove comment

File: packages/indexer/src/jobs/websocket-events/utils.ts (hunk 1 above)
Comment (nofir) [reply packages/indexer/src/jobs/websocket-events/utils.ts:6]: Might be simpler with the mocking but if not, then i would just use the existing one

**Review by nofir — CHANGES_REQUESTED** [review]

**Review by nofir — DISMISSED** [review]

**Review by ipeleg — DISMISSED** [review]

**Comment by georgeroman** [comment]
@nofir or @ipeleg Asking for on last round of review before merging this

**Review by ipeleg — APPROVED** [review]

**Pull request merged**